### ISSUE TITLE
Allow all opam URLs in `opam-repositories` section

### ISSUE THREAD

**Issue by dra27** [ocaml/setup-ocaml#649]
While working on #647, I attempted to add a repository `git+file://D:\\opam-repo#branch` which is a valid opam URL but which causes an exception with setup-ocaml ("Error: Invalid URL" in "Retrieve the opam download cache"). I have a recollection that fully specified opam URLs have caused problems in the past (e.g. `git+https://github.com/ocaml/opam-repository`). The issue looks like it's in `composeOpamDownloadCacheKeys`:
https://github.com/ocaml/setup-ocaml/blob/b0fb36ad1edce66c14923fc242dd0bb3255c0a5e/src/setup-ocaml/cache.ts#L96-L100
when trying to create a URL object.

**Comment by smorimoto** [comment]
So it seems to make more sense here not to create a URL object, but simply to pass it, right?

**Comment by dra27** [comment]
I guess so - I wasn't entirely sure what existing code was trying to do. Is it trying to remove the protocol from the cache key?

**Comment by dra27** [comment]
(this was non-urgent for me, too - it was easy to work-around, and I was setting up an unusual workflow to test something)

**Comment by psafont** [comment]
This issue also happens for local opam repositories: https://github.com/psafont/xs-opam/actions/runs/4435741344/jobs/7783282109
```yaml
      - name: Use ocaml
        uses: ocaml/setup-ocaml@v2
        with:
          ocaml-compiler: "4.08.1"
          opam-repositories: |
            xs-opam: "."
          opam-disable-sandboxing: true
          opam-pin: false
```

I've encountered this because I'm trying to convert a legacy workflow using the ocaml's docker images to setup-ocaml. I don't think I can use v1 (where local repos work fine) as I need to disable opam's sandboxing.

**Comment by smorimoto** [comment]
Fixed in #656
